src/sidebar/index.js

```diff
@@ -182,7 +184,9 @@ function startApp(config) {
   container.run(setupFrameSync);
 
   // Render the UI.
-  const appEl = document.querySelector('hypothesis-app');
+  const appEl = /** @type {HTMLElement} */ (document.querySelector(
+    'hypothesis-app'
+  ));
   render(
     <ServiceContext.Provider value={container}>
```

Comment:
This blows up here, I have not looked too deeply into this yet,  but I suspect it has to do with our Injector type.

Reply:
Ah, interesting. The error here is:

```
src/sidebar/index.js(191,30): error TS2322: Type 'Injector' is not assignable to type '{ get(service: any): never; }'.
  The types returned by 'get(...)' are incompatible between these types.
    Type 'any' is not assignable to type 'never'.
```

Breaking that down, it is complaining that the `value` prop is expecting an object with a `get` method that never returns (that's what the `never` type means), but the object you're passing (`container`) has a `get` method that returns _something_ (`any`).

So the problem here is that `ServiceContext` has the wrong type. Looking at the definition, the reason becomes apparent:

```js
const fallbackInjector = {
  get(service) {
    throw new Error(
      `Missing ServiceContext provider to provide "${service}" prop`
    );
  },
};
...
export const ServiceContext = createContext(fallbackInjector);
```

`ServiceContext`'s type is inferred from `fallbackInjector` where `get` never returns. The fix is to override the type:

```diff
diff --git a/src/sidebar/util/service-context.js b/src/sidebar/util/service-context.js
index a9ef08fce..2dfd0d3bf 100644
--- a/src/sidebar/util/service-context.js
+++ b/src/sidebar/util/service-context.js
@@ -12,9 +12,15 @@
  * @typedef {import("redux").Store} Store
  */
 
+/**
+ * @typedef ServiceProvider
+ * @prop {(serviceName: string) => any} get
+ */
+
 import { createContext, createElement } from 'preact';
 import { useContext } from 'preact/hooks';
 
+/** @type {ServiceProvider} */
 const fallbackInjector = {
   get(service) {
     throw new Error(
```